### PR TITLE
Report stats for tdatas_mtx and prof_dump_mtx 

### PR DESCRIPTION
**Pull request by yinan1048576**

### CODE REVIEW

**Comment by yinan1048576** [comment]
The stats output is confirmed to reflect the info of the additional mutexes. The only thing is that if application requests non-json format and relies on the line numbers for interpreting the output, the lines will be shifted due to this change which may cause some issue.

**Comment by interwq** [comment]
Can you paste some output from `malloc_stats`? Also, if we want to include these in the output, we will need to document them, similar to http://jemalloc.net/jemalloc.3.html#stats.mutexes.prof
that also means we better name the two new counters, which is hard for sure.

**Comment by interwq** [comment]
> if application requests non-json format and relies on the line numbers for interpreting the output

Don't worry about that case; it's not part of the API we are committing to.

**Comment by yinan1048576** [comment]
The two new counters are named `prof_tdatas` and `prof_dump`, and the original prof counter is kept as `prof`.

Example output (JSON format):
```
"stats": {
    …
    "mutexes": {
        "background_thread": {
            "num_ops": 4,
            "num_wait": 4,
            "num_spin_acq": 0,
            "num_owner_switch": 0,
            "total_wait_time": 0,
            "max_wait_time": 0,
            "max_num_thds": 0
        },
        "ctl": {
            "num_ops": 2,
            "num_wait": 2,
            "num_spin_acq": 0,
            "num_owner_switch": 0,
            "total_wait_time": 0,
            "max_wait_time": 0,
            "max_num_thds": 0
        },
        "prof": {
            "num_ops": 0,
            "num_wait": 0,
            "num_spin_acq": 0,
            "num_owner_switch": 0,
            "total_wait_time": 0,
            "max_wait_time": 0,
            "max_num_thds": 0
        },
        "prof_tdatas": {
            "num_ops": 0,
            "num_wait": 0,
            "num_spin_acq": 0,
            "num_owner_switch": 0,
            "total_wait_time": 0,
            "max_wait_time": 0,
            "max_num_thds": 0
        },
        "prof_dump": {
            "num_ops": 0,
            "num_wait": 0,
            "num_spin_acq": 0,
            "num_owner_switch": 0,
            "total_wait_time": 0,
            "max_wait_time": 0,
            "max_num_thds": 0
        }
    }
}
```
Example output (text format):
```

                           n_lock_ops (#/sec)       n_waiting (#/sec)       n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns   max_n_thds
background_thread                  32      32               0       0                0       0               1       1               0         0               0            0
ctl                                44      44               0       0                0       0               1       1               0         0               0            0
prof                               31      31               0       0                0       0               1       1               0         0               0            0
prof_tdatas                        17      17               0       0                0       0               1       1               0         0               0            0
prof_dump                          16      16               0       0                0       0               1       1               0         0               0            0
```
Will add some more documentations shortly.

**Comment by interwq** [comment]
The ‘prof_tdatas’ is probably not an ideal name; more verbose should be better here, e.g. prof_thd_data?

**Comment by yinan1048576** [comment]
Changed the name to `prof_thds_data`. (`thds` instead of `thd` since `tdatas` is global for all `tdata`s.)

File: doc/jemalloc.xml.in [hunk 1]
```diff
@@ -2509,6 +2509,30 @@ struct extent_hooks_s {
         counters</link>.</para></listitem>
       </varlistentry>
 
+      <varlistentry id="stats.mutexes.prof_tdatas">
+        <term>
+          <mallctl>stats.mutexes.prof_thds_data.{counter}</mallctl>
+	  (<type>counter specific type</type>) <literal>r-</literal>
+          [<option>--enable-stats</option>]
+        </term>
+	<listitem><para>Statistics on <varname>prof</varname> threads data mutex
```
Comment (interwq) [inline doc/jemalloc.xml.in]: We probably want to have better explanation, given that we now have 3 prof related locks. That can happen later after checking some production data though.

**Comment by yinan1048576** [comment]
Rebase.

**Pull request merged**